### ISSUE TITLE
Test Flake: istio.io/istio/pilot/pkg/proxy TestApplyThrice

### ISSUE THREAD

**Issue by ozevren** [istio/istio#12752]
https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/12247/istio-unit-tests/22603

**Comment by howardjohn** [comment]
This is still an issue

**Comment by ozevren** [comment]
@andraxylia @rshriram please redirect as appropriate.

**Comment by howardjohn** [comment]
Digging deeper - This commonly fails with 

```
--- PASS: TestApplyThrice (0.00s)
2019-05-02T02:00:14.274025Z    info    Epoch 0 starting
panic: Fail in goroutine after TestApplyThrice has completed

goroutine 37 [running]:
testing.(*common).Fail(0xc42018c000)
    /usr/local/go/src/testing/testing.go:509 +0x134
testing.(*common).Errorf(0xc42018c000, 0x806575, 0x1f, 0x0, 0x0, 0x0)
    /usr/local/go/src/testing/testing.go:585 +0x83
istio.io/istio/pilot/pkg/proxy.TestApplyThrice.func1(0x76d280, 0xc420168070, 0x0, 0xc4201b2060, 0xc42016ef30, 0x71d83f)
    /home/prow/go/src/istio.io/istio/pilot/pkg/proxy/agent_test.go:135 +0x113
istio.io/istio/pilot/pkg/proxy.TestProxy.Run(0xc42019e000, 0xc420186030, 0x0, 0x76d280, 0xc420168070, 0x0, 0xc4201b2060, 0x76c480, 0xc4200266a0)
    /home/prow/go/src/istio.io/istio/pilot/pkg/proxy/agent_test.go:41 +0x4e
istio.io/istio/pilot/pkg/proxy.(*agent).runWait(0xc4201a4000, 0x76d280, 0xc420168070, 0x0, 0xc4201b2060)
    /home/prow/go/src/istio.io/istio/pilot/pkg/proxy/agent.go:299 +0xe3
created by istio.io/istio/pilot/pkg/proxy.(*agent).reconcile
    /home/prow/go/src/istio.io/istio/pilot/pkg/proxy/agent.go:293 +0x24b
FAIL    istio.io/istio/pilot/pkg/proxy    0.008s
```

So a goroutine calls `.Error` *after* the test completes. If you add some delay you'll see it will just fail like normal with `go test ./pilot/pkg/proxy/ -run ApplyThrice -count 100`.

**Comment by crazyxy** [comment]
There is a race condition between `terminate()` and code apply configuration changes. Add a 10ms delay after sending configuration change message.

**Comment by howardjohn** [comment]
This was fixed in #13827 verified in test grid
